streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java

```diff
@@ -35,15 +35,21 @@
 
     private static final Logger log = LoggerFactory.getLogger(StickyTaskAssignor.class);
     private final Map<ID, ClientState> clients;
-    private final Set<TaskId> taskIds;
+    private final Set<TaskId> allTaskIds;
+    private final Set<TaskId> standbyTaskIds;
     private final Map<TaskId, ID> previousActiveTaskAssignment = new HashMap<>();
     private final Map<TaskId, Set<ID>> previousStandbyTaskAssignment = new HashMap<>();
     private final TaskPairs taskPairs;
 
-    public StickyTaskAssignor(final Map<ID, ClientState> clients, final Set<TaskId> taskIds) {
+    public StickyTaskAssignor(final Map<ID, ClientState> clients,
+                              final Set<TaskId> allTaskIds,
+                              final Set<TaskId> standbyTaskIds) {
         this.clients = clients;
-        this.taskIds = taskIds;
-        taskPairs = new TaskPairs(taskIds.size() * (taskIds.size() - 1) / 2);
+        this.allTaskIds = allTaskIds;
+        this.standbyTaskIds = standbyTaskIds;
+
+        final int maxPairs = standbyTaskIds.size() * (standbyTaskIds.size() - 1) / 2;
```

Comment:
This is the one line I'd like some feedback on in particular; if I understand the `TaskPairs` class/purpose correctly, we should change the `maxPairs` to use the number of (distinct) standby tasks, which will always be smaller than the number of total tasks. ie there may be some "unpaired" active tasks, but that's fine since they won't have standbys to begin with

Reply:
The taskPairs is just use as a heuristic to avoid having (active-1, standby-2) and (active-2, standby-1) since if these two instances failed at the same time we lose both tasks: having as large num.pairs as possible means that such case would not likely happen. So if an active task does not have a corresponding standby, it would still appear in some pairs, not sure if that means your new math is correct or not? E.g. let's say we have three tasks A, B, C and A does not have standby while others do have two (num.replicas == 2). An assignment of

(active-A, standby-B, standby-C), (active-B, standby-C), (active-C, standby-B)

would still generates three pairs AB, AC, BC, right?

In any ways, the maxPairs is just used for initializing the hash-set and decide whether we can exit early, so I think setting it to larger values would not matter to much.

Reply:
Yeah, I think it should still be based off of `allTasks.size()` -- I thought only active-standby pairs count but it's actually any unique/new pair. I'll put it back